src/app/portlets/content-types/fields/content-type-fields-drop-zone/content-type-fields-drop-zone.component.ts

```diff
@@ -41,16 +49,106 @@ export class ContentTypeFieldsDropZoneComponent {
     }
 
     /**
-     * Emit the saveField event, this method is call when a Field is dropped into the content type drop zone
+     * Emit the saveField event
+     * @param {Field} fieldToSave
+     * @memberof ContentTypeFieldsDropZoneComponent
      */
-    saveFields(): void {
-        this.saveFieldsEvent.emit(this.getFields());
+    saveFields(fieldToSave: Field): void {
+        let fields = this.getFields();
+        // Needs a better implementation
+        fields.map(field => {
+            if (!field.id && !(fieldToSave.clazz === TAB_DIVIDER.clazz || fieldToSave.clazz === LINE_DIVIDER.clazz)) {
+                field = Object.assign(field, fieldToSave);
+            } else if (field.id === this.formData.id) {
+                field = Object.assign(field, fieldToSave);
+            }
+
+            return field;
+        });
+
+        this.saveFieldsEvent.emit(fields);
+        this.cancelDialog();
+    }
+
+    /**
+     * Get the field to be edited
+     * @param {Field} fieldToEdit
+     * @memberof ContentTypeFieldsDropZoneComponent
+     */
+    editField(fieldToEdit: Field): void {
+        let fields = this.getFields();
+        // Needs a better implementation
+        fields.map((field) => {
+            if (fieldToEdit.id === field.id) {
+                this.formData = fieldToEdit;
+            }
+
+            return field;
+        });
+
+        this.showDialog();
+    }
+
+    /**
+     * Set the field to be edited
+     * @memberof ContentTypeFieldsDropZoneComponent
+     */
+    setDroppedField(): void {
+        let fields = this.getFields();
+        // Needs a better implementation
+        fields.map(field => {
+            if (!field.id && !(field.clazz === TAB_DIVIDER.clazz || field.clazz === LINE_DIVIDER.clazz)) {
+                this.formData = field;
+            }
+        });
     }
 
+    /**
+     * Show the properties field dialog
+     * @memberof ContentTypeFieldsDropZoneComponent
+     */
+    showDialog(): void {
```

Comment:
I think that instead or having2 methods to handle thid, you can create one:

```
toggleDialog() {
    this.displayDialog = !this.displayDialog;
}
```

If not... please use verbs: show/hide.